### DISCUSSION TITLE
How to connect Dagster storage to Google Cloud SQL

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8550] [Q&A]
Hi all. I am testing deploying Dagster via Google Cloud Run. I have a Cloud SQL PostgreSQL instance and Dagster runs well when I run it locally using the instance's public IP as my host for PostgresRunStorage, PostgresEventLogStorage, and PostgresScheduleStorage. For production via Cloud Run, I want my Cloud Run instance to use the [built-in connector for Cloud SQL](https://cloud.google.com/sql/docs/postgres/connect-run#connecting_to). This provides a unix socket that should be used as the host (ie. `/cloudsql/GCP_PROJECT:us-central1:CLOUD_SQL_INSTANCE_NAME`). Using that unix socket path in the host field does not work. I believe that's because I need to use a [different format](https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#unix-domain-connections) for the connection string than what I see [being used](https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-postgres/dagster_postgres/utils.py#L61). I am looking for guidance on how I should proceed from here. Thank you!

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1634954649045100?thread_ts=1634954649.045100&cid=C01U954MEER).*

**Comment by yuhan** [comment]
You can also construct the url yourself and set it as ‘postgres_url’, see one of the examples here: https://docs.dagster.io/deployment/dagster-instance#postgres-storage
